src/cljs/rems/application.cljs

```diff
@@ -84,49 +85,42 @@
 
 ;;;; State
 
-(rf/reg-sub ::application (fn [db _] (::application db)))
+(rf/reg-sub ::application-id (fn [db _] (::application-id db)))
+(rf/reg-sub ::application (fn [db [_ k]] (get-in db [::application (or k :data)])))
 (rf/reg-sub ::edit-application (fn [db _] (::edit-application db)))
 
 (rf/reg-event-fx
  ::enter-application-page
  (fn [{:keys [db]} [_ id]]
-   {:db (dissoc db ::application ::edit-application ::attachment-success)
+   {:db (-> db
+            (assoc ::application-id id)
+            (dissoc ::application ::edit-application ::attachment-success))
     :dispatch [::fetch-application id]}))
 
 (rf/reg-event-fx
  ::fetch-application
- (fn [_ [_ id]]
-   (fetch (str "/api/applications/" id)
-          {:handler #(rf/dispatch [::fetch-application-result %])})
-   {}))
-
-(rf/reg-event-db
- ::fetch-application-result
- (fn [db [_ application]]
-   (assoc db
-          ::application application
-          ::edit-application {:field-values (->> (get-in application [:application/form :form/fields])
-                                                 (map (juxt :field/id :field/value))
-                                                 (into {}))
-                              :show-diff {}
-                              :validation-errors nil})))
-
-(rf/reg-event-fx
- ::reload-application-page
  (fn [{:keys [db]} [_ id]]
-   {::reload-application id}))
-
-(rf/reg-fx
- ::reload-application
- (fn [id]
    (fetch (str "/api/applications/" id)
-          {:handler #(rf/dispatch [::reload-application-result %])})))
+          {:handler #(rf/dispatch [::fetch-application-result %])
+           :error-handler (comp #(rf/dispatch [::fetch-application-result nil])
+                                (flash-message/default-error-handler :top (text :t.applications/application)))})
+   {:db (update db ::application fetcher/started)}))
+
+(defn- initialize-edit-application [db]
+  (let [application (get-in db [::application :data])
```

Comment:
not sure I like the intermediate :data key

how about storing the fetch status under `::application ::fetch`, or perhaps under `::fetch-application` or something

Reply:
after discussion, let's just go with this for now and see if we can come up with something better later